### PR TITLE
nixos/tests: add mastodon test

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Add mastodon test.
cc @erictapen @happy-river @petabyteboy

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
cc @erictapen @ajs124 @aanderse @pingiun

**Comment by erictapen** [comment]
This looks great, thanks a lot! I'll try to build it today or at the latest tomorrow.

**Comment by Izorkin** [comment]
@erictapen needed help to update build package with 'fetchYarnDeps'.

**Comment by erictapen** [comment]
@Izorkin What's the motivation for using fetchYarnDeps? Is this related to this PR? If not please open a separate PR for it.

File: nixos/tests/web-apps/mastodon.nix [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    mkdir -p $out
+    echo insecure-root-password > $out/root-password-file
+    echo insecure-intermediate-password > $out/intermediate-password-file
+    ${pkgs.step-cli}/bin/step certificate create "Example Root CA" $out/root_ca.crt $out/root_ca.key --password-file=$out/root-password-file --profile root-ca
+    ${pkgs.step-cli}/bin/step certificate create "Example Intermediate CA 1" $out/intermediate_ca.crt $out/intermediate_ca.key --password-file=$out/intermediate-password-file --ca-password-file=$out/root-password-file --profile intermediate-ca --ca $out/root_ca.crt --ca-key $out/root_ca.key
+  '';
+in
+{
+  name = "mastodon";
+  meta.maintainers = with pkgs.lib.maintainers; [ izorkin ];
```
Comment (erictapen) [inline nixos/tests/web-apps/mastodon.nix:13]: Could you add myself to that list? I'd gladly help with it.
Also feel free to add yourself as maintainer to the mastodon module and package!

**Comment by Izorkin** [comment]
> What's the motivation for using fetchYarnDeps? Is this related to this PR? If not please open a separate PR for it.

Not related to this PR.

**Review by erictapen — APPROVED** [review]
Looks great, had only two minor nits about it.

**Comment by Izorkin** [comment]
Update PR.

**Pull request merged**